src/NuGet.Services.AzureSearch/DownloadTransferrer.cs

```diff
@@ -0,0 +1,123 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using NuGet.Services.AzureSearch.AuxiliaryFiles;
+
+namespace NuGet.Services.AzureSearch
+{
+    public class DownloadTransferrer : IDownloadTransferrer
+    {
+        private readonly IAuxiliaryFileClient _auxiliaryFileClient;
+        private readonly IDatabaseAuxiliaryDataFetcher _databaseFetcher;
+        private readonly ILogger<DownloadTransferrer> _logger;
+
+        public DownloadTransferrer(
+            IAuxiliaryFileClient auxiliaryFileClient,
+            IDatabaseAuxiliaryDataFetcher databaseFetcher,
+            ILogger<DownloadTransferrer> logger)
+        {
+            _auxiliaryFileClient = auxiliaryFileClient ?? throw new ArgumentException(nameof(auxiliaryFileClient));
+            _databaseFetcher = databaseFetcher ?? throw new ArgumentNullException(nameof(databaseFetcher));
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        }
+
+        public async Task<DownloadTransferResult> GetTransferChangesAsync(DownloadData downloads)
+        {
+            // Downloads are transferred from a "from" package to one or more "to" packages.
+            // The "outgoingTransfers" maps "from" packages to their corresponding "to" packages.
+            _logger.LogInformation("Fetching new popularity transfer data from gallery database.");
+            var outgoingTransfers = await _databaseFetcher.GetPackageIdToPopularityTransfersAsync();
+
+            return await GetTransferChangesAsync(
+                downloads,
+                outgoingTransfers);
+        }
```

Comment:
`GetTransferChangesAsync` and `GetUpdatedTransferChangesAsync` are very similar right now, but they will diverge in a subsequent change.